front-end/src/components/Reactionbar/ReactionButton.tsx

```diff
@@ -12,46 +11,21 @@ import {
 	useAddPostReactionMutation,
 	useAddCommentReactionMutation,
 	useDeletePostReactionMutation,
-	useDeleteCommentReactionMutation,
-
-	DiscussionPostAndCommentsQueryVariables,
-	ProposalPostAndCommentsQueryVariables,
-	ReferendumPostAndCommentsQueryVariables,
-	MotionPostAndCommentsQueryVariables,
-	TreasuryProposalPostAndCommentsQueryVariables,
-
-	DiscussionPostAndCommentsQuery,
-	ProposalPostAndCommentsQuery,
-	ReferendumPostAndCommentsQuery,
-	MotionPostAndCommentsQuery,
-	TreasuryProposalPostAndCommentsQuery
+	useDeleteCommentReactionMutation
 } from '../../generated/graphql';
 
 export interface ReactionButtonProps {
 	className?: string
-	reactionId: number
 	reaction: string
 	count: number
 	userIds: number[]
 	postId?: number
 	commentId?: string
-	refetch?: (variables?:
-		ReferendumPostAndCommentsQueryVariables |
-		DiscussionPostAndCommentsQueryVariables |
-		ProposalPostAndCommentsQueryVariables |
-		MotionPostAndCommentsQueryVariables |
-		TreasuryProposalPostAndCommentsQueryVariables |
-		undefined) =>
-		Promise<ApolloQueryResult<TreasuryProposalPostAndCommentsQuery>> |
-		Promise<ApolloQueryResult<MotionPostAndCommentsQuery>> |
-		Promise<ApolloQueryResult<ReferendumPostAndCommentsQuery>> |
-		Promise<ApolloQueryResult<ProposalPostAndCommentsQuery>> |
-		Promise<ApolloQueryResult<DiscussionPostAndCommentsQuery>>
+	refetch?: any
```

Comment:
Let's type this correctly with
```js
(variables?: PostReactionsQueryVariables | CommentReactionsQueryVariables | undefined) => Promise<ApolloQueryResult<PostReactionsQuery>> | Promise<ApolloQueryResult<CommentReactionsQuery>>
```

Reply:
I think you forgot this :)

Reply:
this is not working. Typescript complaining

Reply:
I didn't check 100%, just giving a tip. This types are given by VSC, so I guess we can figure something one.

Reply:
Failed to compile.

/home/nikhil/parity/polkassembly/front-end/src/components/Reactionbar/CommentReactionBar.tsx
TypeScript error in /home/nikhil/parity/polkassembly/front-end/src/components/Reactionbar/CommentReactionBar.tsx(62,7):
No overload matches this call.
  Overload 1 of 2, '(props: Pick<Pick<ReactionButtonProps, "reaction" | "count" | "refetch" | "className" | "userIds" | "postId" | "commentId"> & Partial<Pick<ReactionButtonProps, never>>, "reaction" | ... 5 more ... | "commentId"> & { ...; } & { ...; }): ReactElement<...>', gave the following error.
    Type '(variables?: CommentReactionsQueryVariables | undefined) => Promise<ApolloQueryResult<CommentReactionsQuery>>' is not assignable to type '(variables?: PostReactionsQueryVariables | CommentReactionsQueryVariables | undefined) => Promise<ApolloQueryResult<PostReactionsQuery>> | Promise<...>'.
      Types of parameters 'variables' and 'variables' are incompatible.
        Type 'PostReactionsQueryVariables | CommentReactionsQueryVariables | undefined' is not assignable to type 'CommentReactionsQueryVariables | undefined'.
          Property 'commentId' is missing in type 'PostReactionsQueryVariables' but required in type 'CommentReactionsQueryVariables'.
  Overload 2 of 2, '(props: StyledComponentPropsWithAs<({ className, reaction, count, userIds, postId, commentId, refetch }: ReactionButtonProps) => Element, any, {}, never>): ReactElement<StyledComponentPropsWithAs<({ className, reaction, count, userIds, postId, commentId, refetch }: ReactionButtonProps) => Element, any, {}, never>, string | ... 1 more ... | (new (props: any) => Component<...>)>', gave the following error.
    Type '(variables?: CommentReactionsQueryVariables | undefined) => Promise<ApolloQueryResult<CommentReactionsQuery>>' is not assignable to type '(variables?: PostReactionsQueryVariables | CommentReactionsQueryVariables | undefined) => Promise<ApolloQueryResult<PostReactionsQuery>> | Promise<...>'.  TS2769

    60 | 						reaction={reaction}
    61 | 						commentId={commentId}
  > 62 | 						refetch={refetch}
       | 						^
    63 | 					/>
    64 | 				);
    65 | 			})}

Reply:
fixed by using 2 variables

Reply:
That works but I had a deeper look to understand why we manage to use such types in other components, it's because event though the queries are different, the variable names were the same. In our case, there's `commentId` and `postId`. The thing is, I don't think we use those. I have a branch with this solution.  https://github.com/paritytech/polkassembly/compare/niklabh-reactions-query-seperate...tbaut-types?expand=1
I would test if the migration was working.